libvcx/src/disclosed_proof.rs

```diff
@@ -273,303 +219,108 @@ pub fn build_rev_states_json(credentials_identifiers: &mut Vec<CredInfo>) -> Vcx
+pub fn build_schemas_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
+    let mut rtn: Value = json!({});
 
-        let proof_req = self.proof_request
-            .as_ref()
-            .ok_or(VcxError::from_msg(VcxErrorKind::NotReady, "Cannot get proot request"))?;
+    for ref cred_info in credentials_identifiers {
+        if rtn.get(&cred_info.schema_id).is_none() {
+            let (_, schema_json) = anoncreds::get_schema_json(&cred_info.schema_id)
+                .map_err(|err| err.map(VcxErrorKind::InvalidSchema, "Cannot get schema"))?;
 
-        let indy_proof_req = serde_json::to_string(&proof_req.proof_request_data)
-            .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize proof request: {}", err)))?;
+            let schema_json = serde_json::from_str(&schema_json)
+                .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidSchema, format!("Cannot deserialize schema: {}", err)))?;
 
-        anoncreds::libindy_prover_get_credentials_for_proof_req(&indy_proof_req)
+            rtn[cred_info.schema_id.to_owned()] = schema_json;
+        }
     }
+    Ok(rtn.to_string())
+}
 
-    pub fn build_schemas_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
-        let mut rtn: Value = json!({});
+pub fn build_cred_def_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
+    let mut rtn: Value = json!({});
 
-        for ref cred_info in credentials_identifiers {
-            if rtn.get(&cred_info.schema_id).is_none() {
-                let (_, schema_json) = anoncreds::get_schema_json(&cred_info.schema_id)
-                    .map_err(|err| err.map(VcxErrorKind::InvalidSchema, "Cannot get schema"))?;
+    for ref cred_info in credentials_identifiers {
+        if rtn.get(&cred_info.cred_def_id).is_none() {
+            let (_, credential_def) = anoncreds::get_cred_def_json(&cred_info.cred_def_id)
+                .map_err(|err| err.map(VcxErrorKind::InvalidProofCredentialData, "Cannot get credential definition"))?;
 
-                let schema_json = serde_json::from_str(&schema_json)
-                    .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidSchema, format!("Cannot deserialize schema: {}", err)))?;
+            let credential_def = serde_json::from_str(&credential_def)
+                .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidProofCredentialData, format!("Cannot deserialize credential definition: {}", err)))?;
 
-                rtn[cred_info.schema_id.to_owned()] = schema_json;
-            }
+            rtn[cred_info.cred_def_id.to_owned()] = credential_def;
         }
-        Ok(rtn.to_string())
     }
+    Ok(rtn.to_string())
+}
 
-    pub fn build_cred_def_json(credentials_identifiers: &Vec<CredInfo>) -> VcxResult<String> {
-        let mut rtn: Value = json!({});
-
+pub fn build_requested_credentials_json(credentials_identifiers: &Vec<CredInfo>,
```

Comment:
Let's move these static functions to `proof_utils.rs` (or `disclosed_proof_utils.rs` if it make more sense for these functions)